fhir-server/src/main/java/com/ibm/fhir/server/resources/FHIRResource.java

```diff
@@ -2088,22 +2092,56 @@ public Bundle doSearch(String type, String compartment, String compartmentId,
         }
     }
 
+    /**
+     * Uses the handling property from the server config and the Prefer header from the request to determine which mode the server should use.
+     * 
+     * @throws IllegalArgumentException if the mode is STRICT and the Prefer header contains an invalid handling value.
+     */
     private boolean isLenient(Map<String, String> requestProperties) {
-        boolean lenient = true;
-
-        String handlingStringValue = getHeaderValue(requestProperties, "Prefer", "handling");
-        if ("strict".equals(handlingStringValue)) {
-            lenient = false;
+        boolean isLenient = false; // Assign it to appease the compiler
```

Comment:
to appease the compiler in this case, you need to set isLenient in the default branch of the below, however, I assume Handling.from ensures no other case.

Reply:
correct.  we could either initialize it or create a default block.  neither is actually needed because its an enum and we have cases for all possible values.